src/traversing/findFilter.js

```diff
@@ -15,24 +15,29 @@ function winnow( elements, qualifier, not ) {
 		return jQuery.grep( elements, function( elem, i ) {
 			return !!qualifier.call( elem, i, elem ) !== not;
 		} );
-
 	}
 
+	// Single element
 	if ( qualifier.nodeType ) {
 		return jQuery.grep( elements, function( elem ) {
 			return ( elem === qualifier ) !== not;
 		} );
-
 	}
 
-	if ( typeof qualifier === "string" ) {
-		if ( risSimple.test( qualifier ) ) {
-			return jQuery.filter( qualifier, elements, not );
-		}
+	// Arraylike of elements (jQuery, arguments, Array)
+	if ( typeof qualifier !== "string" ) {
+		return jQuery.grep( elements, function( elem ) {
+			return ( indexOf.call( qualifier, elem ) > -1 ) !== not;
+		} );
+	}
 
-		qualifier = jQuery.filter( qualifier, elements );
+	// Simple selector that can be filtered directly, removing non-Elements
+	if ( risSimple.test( qualifier ) ) {
```

Comment:
(Side note, this can be addressed in a different PR but I wanted to be sure this seemed correct.)

This test for a simple selector needs to be there for the "not" case mainly because `jQuery.filter()` always removes text/comment nodes. However, it should be safe to always take this branch if `!not` correct? It seems this appeared in the `.filter()` path in 87869b8b81380617cfad8788799ee40de0e3e9c4 when it was DRYed out.